models/address_book.rb

```diff
@@ -18,4 +19,18 @@ def add_entry(name, phone, email)
 
     @entries.insert(index, Entry.new(name, phone, email))
   end
+
+  def remove_entry(name, phone, email)
+    # using index where I could have set a delete_entry variable
+    index = 0
+
+    @entries.each do |entry|
+      if name == entry.name && phone = entry.phone_number && email == entry.email
+        break
+      end
+      index += 1
+    end
+    # could have said @entries.delete(delete_entry)
```

Comment:
You can still change this and remove the `index` variable :+1: See if you can get it working with just the `each` block and `entry` :smile: